gcloud/monitoring/test_client.py

```diff
@@ -0,0 +1,325 @@
+        self.assertEqual(descriptor.display_name, DISPLAY_NAME)
+        self.assertEqual(descriptor.description, DESCRIPTION)
+
+        self.assertEqual(len(descriptor.labels), 3)
+        label1, label2, label3 = descriptor.labels
+        self.assertEqual(label1.key, LABEL1['key'])
+        self.assertEqual(label2.key, LABEL2['key'])
+        self.assertEqual(label3.key, LABEL3['key'])
+
+        request, = connection._requested
+        expected_request = {'method': 'GET', 'path': '/' + NAME}
+        self.assertEqual(request, expected_request)
+
+    def test_list_resource_descriptors(self):
+        PATH = 'projects/{project}/monitoredResourceDescriptors/'.format(
+            project=PROJECT)
+
+        TYPE1 = 'custom.googleapis.com/resource-1'
+        DESCRIPTION1 = 'This is the first resource.'
+        NAME1 = PATH + TYPE1
+        RESOURCE_DESCRIPTOR1 = {
+            'name': NAME1,
+            'type': TYPE1,
+            'description': DESCRIPTION1,
+        }
+
+        TYPE2 = 'custom.googleapis.com/resource-2'
+        DESCRIPTION2 = 'This is the second resource.'
+        NAME2 = PATH + TYPE2
+        RESOURCE_DESCRIPTOR2 = {
+            'name': NAME2,
+            'type': TYPE2,
+            'description': DESCRIPTION2,
+        }
+
+        RESPONSE = {
+            'resourceDescriptors':
+                [RESOURCE_DESCRIPTOR1, RESOURCE_DESCRIPTOR2],
+        }
+
+        # This test is identical to TestResourceDescriptor.test_list()
+        # except for the following three lines.
+        client = self._makeOne(project=PROJECT, credentials=_Credentials())
+        connection = client.connection = _Connection(RESPONSE)
+        descriptors = client.list_resource_descriptors()
+
+        self.assertEqual(len(descriptors), 2)
+        descriptor1, descriptor2 = descriptors
+
+        self.assertEqual(descriptor1.name, NAME1)
+        self.assertEqual(descriptor1.type, TYPE1)
+        self.assertEqual(descriptor1.description, DESCRIPTION1)
+
+        self.assertEqual(descriptor2.name, NAME2)
+        self.assertEqual(descriptor2.type, TYPE2)
+        self.assertEqual(descriptor2.description, DESCRIPTION2)
+
+        request, = connection._requested
+        expected_request = {'method': 'GET', 'path': '/' + PATH,
+                            'query_params': {}}
+        self.assertEqual(request, expected_request)
+
+
+class _Credentials(object):
+
+    _scopes = None
+
+    @staticmethod
+    def create_scoped_required():
+        return True
+
+    def create_scoped(self, scope):
+        self._scopes = scope
+        return self
+
+
+class _Connection(object):
+
+    def __init__(self, *responses):
+        self._responses = list(responses)
+        self._requested = []
+
+    def api_request(self, **kwargs):
+        from gcloud.exceptions import NotFound
+        self._requested.append(kwargs)
+        try:
+            return self._responses.pop(0)
+        except IndexError:  # pragma: NO COVER
```

Comment:
I get why you put it there, but none of the tests you added actually traverse this line. So if you'd like to have it as a fallback, add a test which triggers the line and drop the `NO COVER`.